app/index/controller.js

```diff
@@ -209,21 +233,65 @@ export default Controller.extend({
   },
 
   /**
+   * The workdays for the location related to the users active employment
+   *
+   * @property {Number[]} workdays
+   * @public
+   */
+  @computed('session.data.authenticated.user_id')
+  workdays(userId) {
+    return this.store.peekRecord('user', userId).get('activeEmployment.location.workdays')
+  },
+
+  /**
    * The data for the weekly overview
    *
    * @property {Object[]} weeklyOverviewData
    * @public
    */
-  @computed('_allReports.@each.{date,duration}', 'days.[]', 'date')
-  weeklyOverviewData(reports, days, current) {
-    return Array.from({ length: 31 }, (v, k) => moment(current).add(k - 20, 'days')).map((d) => {
+  @computed('_allReports.@each.{duration,date}', '_allAbsences.@each.{duration,date}', 'date')
+  weeklyOverviewData(allReports, allAbsences, date) {
+    let task = this.get('_weeklyOverviewData')
+
+    task.perform(allReports, allAbsences, date)
+
+    return task
+  },
+
+  /**
+   * The task to compute the data for the weekly overview
+   *
+   * @property {EmberConcurrency.Task} _weeklyOverviewData
+   * @private
+   */
+  _weeklyOverviewData: task(function* (allReports, allAbsences, date) {
+    yield timeout(200)
+
+    allReports  = allReports.filter((r)  => !r.get('isDeleted') && !r.get('isNew'))
+    allAbsences = allAbsences.filter((a) => !a.get('isDeleted') && !a.get('isNew'))
+
+    let container = [ ...allReports, ...allAbsences ].reduce((obj, model) => {
+      let d = model.get('date').format('YYYY-MM-DD')
+
+      obj[d] = obj[d] || { reports: [], absences: [] }
+
+      obj[d][`${model.constructor.modelName}s`].push(model.get('duration'))
+
+      return obj
+    }, {})
+
+    return Array.from({ length: 31 }, (v, k) => moment(date).add(k - 20, 'days')).map((d) => {
+      let { reports = [], absences = [] } = container[d.format('YYYY-MM-DD')] || {}
+
       return {
         day: d,
-        active: d.isSame(current, 'day'),
-        worktime: reports.filter((r) => r.get('date').isSame(d, 'day')).reduce((val, cur) => {
-          return val.add(cur.get('duration'))
+        active: d.isSame(date, 'day'),
+        absence: !!absences.length,
+        workday: this.get('workdays').includes(d.isoWeekday()),
+        worktime: [ ...reports, ...absences ].reduce((val, dur) => {
+          return val.add(dur)
```

Comment:
Ah, nice. I thought so because in `app/index/controller.js#135` you wrote it in a mutating style - or is this because it's a duration?